### PR TITLE
hwmv2: Part bcm_vk socs and boards

### PR DESCRIPTION
**Pull request by nordicjm**
*Includes first hwmv2 conversion of a multi-arch soc/board*

    tests: Update board names for hwmv2
    
    Updates tests that use board names which have changed with
    boards v2
    

    boards: arm: bcm958401m2: Convert to v2
    
    Converts the board to hwmv2
    

    boards: arm: bcm958402m2: Convert to v2
    
    Converts the board to hwmv2
    

    soc: bcm_vk: Port to HWMv2
    
    Ports the bcm_vk SoC configuration to hardware model version 2

### CODE REVIEW

File: boards/v2/broadcom/bcm958401m2/board.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: bcm958401m2
+  vendor: Broadcom
+  socs:
+  - name: bcm58400
```
Comment (tejlmand) [inline boards/v2/broadcom/bcm958401m2/board.yml:5]: check the line ending.

File: soc/v2/broadcom/bcm_vk/Kconfig [hunk 2]
```diff
@@ -7,10 +7,11 @@ config SOC_FAMILY_BCMVK
 	bool
```
Comment (tejlmand) [inline soc/v2/broadcom/bcm_vk/Kconfig:7]: cannot place comment on line above, but should this setting:
https://github.com/zephyrproject-rtos/zephyr/blob/fbdb44e26fe5febce19818c8584a863aeb5b9db5/soc/v2/broadcom/bcm_vk/Kconfig#L6-L7

be moved here:
https://github.com/zephyrproject-rtos/zephyr/blob/fbdb44e26fe5febce19818c8584a863aeb5b9db5/soc/v2/broadcom/bcm_vk/Kconfig.soc#L5

as it is referenced here (and other places as well):
https://github.com/zephyrproject-rtos/zephyr/blob/fbdb44e26fe5febce19818c8584a863aeb5b9db5/soc/v2/broadcom/bcm_vk/viper/Kconfig.soc#L6

**Review by tejlmand — CHANGES_REQUESTED** [review]
minor nits, but overall looks good 👍

File: soc/v2/broadcom/bcm_vk/Kconfig (hunk 2 above)
Comment (nordicjm) [reply soc/v2/broadcom/bcm_vk/Kconfig]: Fixed

**Comment by nordicjm** [comment]
@dleach02 @erwango this has a dual-architecture SoC (arm code and arm64 cores) which has been converted to hwmv2, might be a useful reference for your own multi-arch conversions

**Review by tejlmand — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**